### PR TITLE
[HW][TP] Thread Pool

### PR DESCRIPTION
**Pull request by IShirokov1**

### CODE REVIEW

File: ThreadPool/ThreadPool/ThreadPool.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		[&value](std::pair<std::string, task_type> const& elem) {
+			return elem.first == value;
+		});
+
+	if (!g_instance->m_initialized || it != taskQueue.end()) {
+		return false;
+	}
+
+	taskQueue.push_back(std::make_pair(label, task));
+	m_remainingTasks.store(m_remainingTasks + 1);
```
Comment (an-ovsyuhno) [inline ThreadPool/ThreadPool/ThreadPool.cpp:22]:
```suggestion
	m_remainingTasks++;
```

File: ThreadPool/ThreadPool/ThreadPool.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (IShirokov1) [reply ThreadPool/ThreadPool/ThreadPool.h]: Files have been moved

File: ThreadPool/ThreadPool/ThreadPool.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+#include <iostream>
+#include <algorithm>
+
+#include "ThreadPool.h"
+
+bool ThreadPool::queue(const std::string& label, task_type task)
+{
+	std::unique_lock<std::mutex> lock(mutexTQ);
+
+	std::string value = label;
```
[the remark this answers is not in the document]
Comment (IShirokov1) [reply ThreadPool/ThreadPool/ThreadPool.cpp]: Fixed

File: ThreadPool/ThreadPool/Source.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+#include <random>
+#include <string>
+
+std::mutex coutMutex;
+std::mt19937 gen(time(0));
+std::uniform_int_distribution<> uid(0, 3);
+
+int main()
+{
+	auto func = []() {
```
[the remark this answers is not in the document]
Comment (IShirokov1) [reply ThreadPool/ThreadPool/Source.cpp]: Did you mean the tab before the first curly brace? Then fixed

**Pull request closed without merging**